### PR TITLE
Implent JointRegularizationTask in TSID

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This PR:
1. fixes a bug in `Task::subA()` method
2. Implement `JointRegularizationTask` in TSID library

### CODE REVIEW

File: src/TSID/include/BipedalLocomotion/TSID/JointRegularizationTask.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    /**
+     * Set the desired reference trajectory.
+     * @param jointPosition vector containing the desired joint position in radians.
+     * @param jointVelocity vector containing the desired joint velocity in radians per second.
+     * @param jointAcceleration vector containing the desired joint velocity in radians per second square.
+     * @return True in case of success, false otherwise.
+     */
+    bool setReferenceTrajectory(Eigen::Ref<const Eigen::VectorXd> jointPosition,
+                                Eigen::Ref<const Eigen::VectorXd> jointVelocity,
+                                Eigen::Ref<const Eigen::VectorXd> jointAcceleration);
```
Comment (S-Dafarra) [inline src/TSID/include/BipedalLocomotion/TSID/JointRegularizationTask.h:92]: You are not setting trajectories, but rather desired set points.

File: src/TSID/src/JointRegularizationTask.cpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+using namespace BipedalLocomotion::TSID;
+
+bool JointRegularizationTask::initialize(
+    std::weak_ptr<ParametersHandler::IParametersHandler> paramHandler,
+    const System::VariablesHandler& variablesHandler)
+{
+    constexpr std::string_view errorPrefix = "[JointRegularizationTask::initialize] ";
+
+    constexpr std::string_view descriptionPrefix = "JointRegularizationTask Optimal Control "
+                                                   "Element";
```
Comment (S-Dafarra) [inline src/TSID/src/JointRegularizationTask.cpp:22]: I am not sure this adds any detail in the error messages. Also, it is no more an ``Optimal Control Element``.

Comment (GiulioRomualdi) [reply src/TSID/src/JointRegularizationTask.cpp]: Remove here: https://github.com/dic-iit/bipedal-locomotion-framework/pull/170/commits/273400091eb76b53c7e8ebb8d687bb0ff25945ee

File: src/TSID/include/BipedalLocomotion/TSID/JointRegularizationTask.h (hunk 1 above)
Comment (GiulioRomualdi) [reply src/TSID/include/BipedalLocomotion/TSID/JointRegularizationTask.h]: changed here: https://github.com/dic-iit/bipedal-locomotion-framework/pull/170/commits/40b888299f1a7925bac7d85b22fc7680ff88b922

Comment (S-Dafarra) [reply src/TSID/include/BipedalLocomotion/TSID/JointRegularizationTask.h]: Just to mess up with you a little more :grin: ``setSetpoints``sounds funny, ``setReferencePoint``? :grin:

**Review by S-Dafarra — APPROVED** [review]

**Pull request merged**